### PR TITLE
Add Chrome based Edge to the Ruby bindings

### PR DESCRIPTION
**Pull request by twalpole**
- [X ] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

This is the start of adding Chrome based Edge to the Ruby bindings.  We need to decide whether to implement this separately from legacy edge, or to accept some sort of configuration option to switch between legacy and chrome based - thoughts?

### CODE REVIEW

**Comment by twalpole** [comment]
I've updated this to move the edge 18 into EdgeLegacy module, with Edge being an alias for it.  Swapping from Edge/:edge being Legacy to Chrome based should be simple if/when chrome based edge goes release

**Comment by p0deje** [comment]
This is fine by me. I know that Java calls legacy `EdgeHtml` while Python keeps old name and calls a new one `ChromiumEdge` (#7253). We might want to make the naming consistent when other bindings support lands, but that's for the future.

**Comment by twalpole** [comment]
I swapped the legacy edge namespace to EdgeHtml since that is what the engine is called -- thanks for pointing that out @p0deje.  Anyone have issues with me squashing one merging this now?

File: rb/spec/integration/selenium/webdriver/edge_chrome/options_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        it 'passes args correctly' do
+          options.add_argument('--user-agent=foo;bar')
+
+          create_driver!(options: options) do |driver|
+            ua = driver.execute_script 'return window.navigator.userAgent'
+            expect(ua).to eq('foo;bar')
+          end
+        end
+
+        # it 'should be able to run in headless mode with #headless!' do
```
Comment (p0deje) [inline rb/spec/integration/selenium/webdriver/edge_chrome/options_spec.rb]: What about this one?

**Comment by p0deje** [comment]
> Anyone have issues with me squashing and merging this now?
No issues from my side. Thanks for tackling this!

File: rb/spec/integration/selenium/webdriver/edge_chrome/options_spec.rb (hunk 1 above)
Comment (twalpole) [reply rb/spec/integration/selenium/webdriver/edge_chrome/options_spec.rb]: It currently crashes when trying to start headless - not sure if they plan to support a headless mode or not

**Pull request merged**